### ISSUE TITLE
Switching to fullscreen takes too long to change quality (YouTube)

### ISSUE THREAD

**Issue by chylex** [meandavejustice/min-vid#1146]
Hi,

I like having low quality playback for the small window, however most of the times I go into fullscreen because I want to look at something specific in greater detail, which is often impossible because:
1) The quality takes too long to change
2) When I miss what I wanted to see, I have to exit fullscreen to go back in time, so the addon probably thinks it doesn't need to increase quality anymore and the low quality part is already in cache so the quality switch now takes even longer and still misses what I wanted to see

I'd appreciate a way to force quality to change instantly when I enter fullscreen. I know there's an open issue on customizing playback quality, but I figured I'd open a separate issue since the quality adjustment is automatic and not manual.

**Comment by Softvision-CristinaBadescu** [comment]
Hello,

I've managed to reproduce this behavior using the latest Nightly build 59.0a1 (2017-12-11) on Windows 10 x64.
I navigated to YouTube, selected a video, set the quality to "Auto" and send it to the Min Vid player. When I was in full screen mode only after some seconds the quality was adjusted. 

However, this seems to be the YouTube expected behavior for "Auto" quality settings. And it seems also logic since Min Vid takes what setting you throw at him from the YouTube player. Also, because the video starts from scratch, it's somehow logic that the mini player will need to measure and adapt. 

In order to bypass your issue, I would suggest that you manually set once on a YouTube video the quality that you desire and YouTube and Min Vid will always open the videos with that quality if available. This is how YouTube works. If you manually choose a setting, they will require it on every opened video if available. If not, it will downgrade to the closest quality.

Still, after performing investigations on this behavior, I've encountered another issue (see #1147). When you manually change the video resolution from the YouTube player (it doesn't matter if it's better or lower quality), if you rapidly send the video to the Min Vid player, it doesn't respect the changed resolution.
@meandavejustice do you think that the behavior from my bug (#1147) is somehow related or we consider this as normal behavior?

**Comment by chylex** [comment]
For years I've been using Youtube Center to force videos to play in HD, because from experience in the past, YT kept ignoring my settings and forcing Auto quality anyway - so all videos on YT are already forced into HD as soon as I open the page, but this has no effect on MinVid.

I know there used to be an account setting to prefer HD videos, which now seems to be gone, but I think even that was semi-broken and YT was doing whatever it wanted, leading me to use YT Center to fix the myriad annoying issues.

**Comment by SoftVision-PaulOiegas** [comment]
Yes, YouTube removed that account setting because the quality setting available now on the video does exactly that. If you manually choose a value there, like 1080p, all the next videos will be automatically forced to 1080p if available. If the set quality is not available, it will downgrade to the next available without affecting the next videos chosen quality.

Anyway, now that you detailed more your behavior, I think it may have the same root cause as #1147. Probably because of YouTube Center actions, somehow the Min Vid player doesn't get the settings.

Can you try to let the YT video play for like 10 seconds and then send it to Min Vid? I'm curious if it's working. If yes, then both the issues will be fixed when the dev team will land a fix.

**Comment by chylex** [comment]
Sending a video fully loaded in HD into MinVid does not respect the quality setting, however it looks like setting the video quality manually once actually fixed the issue for the current *and* future videos, including those played by clicking the thumbnail icon in the Subscriptions page - meaning I cannot even reproduce the 10 second issue, all videos are immediately played in HD without having to wait (even with YT Center disabled).

**Comment by SoftVision-PaulOiegas** [comment]
Good to know, and glad this helped you. Let us know if you encounter any issues in the future and we'll be happy to help.

Have a nice day!
